applications/LinearSolversApplication/custom_solvers/mkl_smoother_base.cpp

```diff
@@ -172,6 +178,27 @@ MKLSmootherBase<TSparse,TDense>::MakeSystemView(const SparseMatrix& rLhs,
     KRATOS_CATCH("")
 }
 
+template <class TSparse, class TDense>
+bool MKLSmootherBase<TSparse,TDense>::CheckThreadConsistency()
+{
+    const int number_of_threads_mkl = mkl_get_max_threads();
+    const int number_of_threads_used = ParallelUtilities::GetNumThreads();
+    if (number_of_threads_mkl > number_of_threads_used) {
+        KRATOS_WARNING("MKLSmootherBase") << "Setting the number of threads in MKL to adapt to ParallelUtilities::GetNumThreads(): " << number_of_threads_used << " instead of " << number_of_threads_mkl << std::endl;
+        return false;
+    }
+    return true;
+}
+
+template <class TSparse, class TDense>
+void MKLSmootherBase<TSparse,TDense>::EnsureMKLThreadConsistency()
+{
+    if (!CheckThreadConsistency()) {
+        const int number_of_threads_used = ParallelUtilities::GetNumThreads();
+        mkl_set_num_threads(number_of_threads_used);
+    }
+}
+
```

Comment:
there seems to be quite the mess with old and new code, can you please clean up?